tools/test.py

```diff
@@ -127,10 +131,25 @@
                                "Currently set search path: %s"
                        % (toolchain, search_path))
 
+         # boolean, true if test configuration file is for module false if mbed-os interface
+        module_conf = False
+
+        # Assign config file. Precedence: test_config>app_config
+        # TODO: merge configs if both given
+        if options.test_config:
+            config, module_conf = get_test_config(options.test_config, mcu)
+            if not config:
+                args_error(parser, "argument --test-config contains invalid path or identifier")
+        elif not options.app_config:
+            config = TestConfig.get_default_config(mcu)
+        else:
+            config = options.app_config
```

Comment:
The app configuration json file. The affected part of the tools would be build APIs called into by the test tools https://github.com/sarahmarshy/mbed-os/blob/2e161c9a0b6d4a584593de5938739d706b4b0c09/tools/test.py#L227